dataloaden.go

```diff
@@ -63,9 +65,12 @@ func getData(typeName string) (templateData, error) {
 		return templateData{}, fmt.Errorf("cant determine working dir: %s", err.Error())
 	}
 
-	pkgName := strings.Join(parts[:len(parts)-1], ".")
+	pkgData, err := getPackage(wd)
+	if err != nil {
+		return templateData{}, fmt.Errorf("failed to resolve output package name: %s", err.Error())
```

Comment:
could this fall back to the old behavior instead?

Reply:
The fallback is done in `getPackage`, but not if `parser.ParseDir` returns an error (not sure when this exactly happens with `parser.PackageClauseOnly`). Should I ignore this error?

Reply:
yeah handling it by falling back to the directory name seems ok.

Reply:
Done, I've also rebased the commit and updated the example with the changes from the previous PR.